conans/test/unittests/client/rest/downloader_test.py

```diff
@@ -48,12 +49,17 @@ def get(self, *_args, **kwargs):
         headers = {"Content-Length": len(self._data), "Accept-Ranges": "bytes"}
         if match and self._accept_ranges:
             start = int(match.groups()[0])
-            status = 206
-            headers.update({"Content-Length": len(self._data) - start,
-                            "Content-Range": "bytes {}-{}/{}".format(start, len(self._data) - 1,
-                                                                     len(self._data))})
-            assert start <= len(self._data)
-
+            if start < len(self._data):
+                status = 206
+                headers.update({"Content-Length": str(len(self._data) - start),
+                                "Content-Range": "bytes {}-{}/{}".format(start, len(self._data) - 1,
+                                                                         len(self._data))})
+            else:
+                status = 416
+                headers.update({"Content-Length": "0",
+                                "Content-Range": "bytes */{}".format(len(self._data))})
```

Comment:
Thanks very much for contributing tests to this feature!

It seems that this syntax is not accepted by the current regex.

Reply:
Yes. This test should fail with the current implementation in version 1.25 as it models the problem some users are currently experiencing. Some servers respond with status 416, others might provide a response which leads to requests exception:
`requests.exceptions.ContentDecodingError: ('Received response with content-encoding: gzip, but failed to decode it.', error('Error -3 while decompressin
g data: incorrect header check'))`
 
The test should pass with the fix you provided in PR #6996. But that is currently on another branch. What do you suggest?